mesop/server/server.py

```diff
@@ -126,15 +145,34 @@ def generate_data(ui_request: pb.UiRequest) -> Generator[str, None, None]:
         runtime().context().reset_current_node()
 
         result = runtime().context().run_event_handler(ui_request.user_event)
+        has_run_navigate_on_load = False
         for _ in result:
           path = ui_request.path
           for command in runtime().context().commands():
             if command.HasField("navigate"):
               path = command.navigate.url
+              page_config = runtime().get_page_config(path=path)
+              if (
+                page_config
+                and page_config.on_load
+                and not has_run_navigate_on_load
+              ):
+                has_run_navigate_on_load = True
+                result = page_config.on_load(LoadEvent(path=path))
+                # on_load is a generator function then we need to iterate through
+                # the generator object.
+                if result:
+                  for _ in result:
+                    yield from render_loop(
+                      path=path, init_request=True, keep_alive=True
+                    )
+                    runtime().context().set_previous_node_from_current_node()
+                    runtime().context().reset_current_node()
```

Comment:
[question] Ditto here, did you test cases where component diffing could come into play? I feel like that's an area for potential regression. Most likely should be fine. But a good place to have some test coverage.

Reply:
Updated tests in mesop/tests/e2e/on_load_test.ts to be more comprehensive